src/evo/deterministicmns.cpp

```diff
@@ -606,7 +606,7 @@ bool CDeterministicMNManager::ProcessBlock(const CBlock& block, const CBlockInde
             newList.SetHeight(nHeight);
         }
 
-        newList.SetBlockHash(block.GetHash());
```

Comment:
I guess `UndoBlock` is not called same often as `ProcessBlock`, so, performance impact is much less noticeable. 
But I'd like to suggest to update `UndoBlock` too.
```
uint256 blockHash = block.GetHash();
```